lib/loader/find_bundle.js

```diff
@@ -9,30 +14,126 @@ module.exports = findBundles;
  * @return {Array.<moduleName>}
  */
 function findBundles(loader) {
+	// Remove the file: protocol portion of the baseURL because glob doesn't
+	// work with it.
+	var cwd = loader.baseURL.replace("file:", "");
+
+	var globOpts = {
+		cwd: path.normalize(cwd),
+		nodir: true
+	};
+	var patternArray = [];
+	var bundleArray = [];
+
+	// add direcotries.lib to globs current working directory
+	if(loader.npmContext && loader.directories && loader.directories.lib) {
+		globOpts.cwd = path.join(globOpts.cwd, loader.directories.lib);
+	}
+
+
 	if(Array.isArray(loader.bundle)) {
-		return loader.bundle;
-	} else if(typeof loader.bundle !== "string") {
-		return [];
+		patternArray = loader.bundle;
+	}else if(typeof loader.bundle === "string") {
+		patternArray.push(loader.bundle);
+	} else {
+		return patternArray;
 	}
 
-	// Support for globs,
-	// System.bundle = "components/**/*";
-	var pattern = loader.bundle;
-	var bundle = glob(pattern, {
-		cwd: path(loader.baseURL)
-	}).map(minusJS);
+	patternArray.forEach(function (pattern) {
+		if(!endsWith(pattern, ".js")) {
+			pattern = pattern+".js";
+		}
 
-	return bundle;
-}
+		// we have a npm app
+		if(loader.npmContext){
+			var app = loader.npmContext.pkgInfo[0];
+
+			if(isNpm(pattern)) {
+				pattern = parseNpmModulename(pattern).modulePath;
+
+			}else if(startsWith("~/", pattern)) {
+				// remove ~/
+				pattern = pattern.substr(2);
+
+			}else if(startsWith(app.name+"/", pattern)){
+				pattern = pattern.substr((app.name+"/").length);
+			}
+		}
+
+
+		bundleArray = _.union(bundleArray, glob.sync(pattern, globOpts).map(function (filepath) {
+			var file = path.parse(filepath);
+			return ((loader.npmContext) ? "~/" : "") + path.join(file.dir, file.name);
+		}));
+	});
 
-// Remove the file: protocol portion of the baseURL because glob doesn't
-// work with it.
-function path(baseURL) {
-	return baseURL.replace("file:", "");
+	return bundleArray;
 }
 
-// Remove the .js extension to make these proper module names.
-function minusJS(name) {
-	var idx = name.indexOf(".js");
-	return idx === -1 ? name : name.substr(0, idx);
+function endsWith(str, suffix) {
+	return str.indexOf(suffix, str.length - suffix.length) !== -1;
 }
+function startsWith(starts, path) {
+	return path.substr(0,starts.length) === starts;
+}
+function isNpm(moduleName){
```

Comment:
you can require npmUtils and get all of this stuff: `var npmUtils = require("steal/ext/npm-utils");`